src/main/java/com/worksap/nlp/sudachi/LatticeNodeImpl.java

```diff
@@ -134,4 +136,30 @@ public String toString() {
         return String.format("%d %d %s(%d) %d %d %d %d", getBegin(), getEnd(), surface, wordId, pos, leftId, rightId,
                 cost);
     }
+
+    void appendSplitted(List<LatticeNode> result, Tokenizer.SplitMode mode) {
+        if (mode == Tokenizer.SplitMode.A) {
+            appendSplitted(result, getWordInfo().getAunitSplit());
+        } else if (mode == Tokenizer.SplitMode.B) {
+            appendSplitted(result, getWordInfo().getBunitSplit());
+        } else {
+            result.add(this);
+        }
+    }
+
+    void appendSplitted(List<LatticeNode> result, int[] ids) {
+        if (ids.length <= 1) {
+            result.add(this);
```

Comment:
I do not understand this comment.

0-length splits means that the current token is not splittable and should be added as it is.
1-length splits mean... that the current token should be replaced with another one? Is this behavior actually used?

Reply:
It is possible to put something different from `this` in the ids, and while we don't usually do that, if we do, it would be an incomprehensible bug, so it should be checked.

Reply:
When someone only looks at the signature of this method, they would not expect it to put `this` in if the length of the `ids` was 1, regardless of the contents of the `ids`.

Reply:
Handling 1 case now as rewriting of the morpheme if it has a different word id.

Reply:
Thank you!